From eff3c0ee53a348ac30641134d527fb5d1abbe448 Mon Sep 17 00:00:00 2001
From: Tobias <ruck.tobias@gmail.com>
Date: Mon, 27 Jun 2022 01:39:31 +0200
Subject: [PATCH 164/196] Fix off-by-one bug in `GetFirstUndoOffset` (#259)

This would calculate the wrong undo pos for blocks that have exactly 253 transactions.
---
 src/nng_interface/nng_interface.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/nng_interface/nng_interface.cpp b/src/nng_interface/nng_interface.cpp
index 72f712121..48924f514 100644
--- a/src/nng_interface/nng_interface.cpp
+++ b/src/nng_interface/nng_interface.cpp
@@ -399,7 +399,7 @@ size_t GetFirstBlockTxOffset(const CBlock &block, const CBlockIndex *pindex) {
 }
 
 size_t GetFirstUndoOffset(const CBlock &block, const CBlockIndex *pindex) {
-    return pindex->nUndoPos + GetSizeOfCompactSize(block.vtx.size());
+    return pindex->nUndoPos + GetSizeOfCompactSize(block.vtx.size() - 1);
 }
 
 flatbuffers::Offset<NngInterface::Block>
-- 
2.39.5 (Apple Git-154)

